Explore crashes when rotating screen

**Steps to reproduce:** 

1. Open "Explore"
2. Rotate screen

**Add System logs:**
```
06-01 21:42:24.836 15945-15945/fr.free.nrw.commons.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: fr.free.nrw.commons.debug, PID: 15945
    java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons.debug/fr.free.nrw.commons.category.CategoryImagesActivity}: java.lang.ClassCastException: fr.free.nrw.commons.category.CategoryImagesListFragment cannot be cast to fr.free.nrw.commons.media.MediaDetailPagerFragment
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2684)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2751)
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4556)
        at android.app.ActivityThread.-wrap19(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1502)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6186)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
     Caused by: java.lang.ClassCastException: fr.free.nrw.commons.category.CategoryImagesListFragment cannot be cast to fr.free.nrw.commons.media.MediaDetailPagerFragment
        at fr.free.nrw.commons.category.CategoryImagesActivity.onCreate(CategoryImagesActivity.java:64)
        at android.app.Activity.performCreate(Activity.java:6684)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2637)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2751) 
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4556) 
        at android.app.ActivityThread.-wrap19(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1502) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6186) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779) 
```
 
 **Commons app version:** 

2.7.1-debug-browse-commons-via-app

--- a/app/src/main/java/fr/free/nrw/commons/category/CategoryImagesActivity.java
+++ b/app/src/main/java/fr/free/nrw/commons/category/CategoryImagesActivity.java
@@ -59,11 +59,6 @@ public class CategoryImagesActivity
         supportFragmentManager = getSupportFragmentManager();
         setCategoryImagesFragment();
         supportFragmentManager.addOnBackStackChangedListener(this);
-        if (savedInstanceState != null) {
-            mediaDetails = (MediaDetailPagerFragment) supportFragmentManager
-                    .findFragmentById(R.id.fragmentContainer);
-
-        }
         requestAuthToken();
         initDrawer();
         setPageTitle();
@@ -113,6 +108,16 @@ public class CategoryImagesActivity
             supportFragmentManager.executePendingTransactions();
         }
         mediaDetails.showImage(i);
+    }
+
+    @Override
+    protected void onResume() {
+        if (supportFragmentManager.getBackStackEntryCount()==1){
+            //FIXME: Temporary fix for screen rotation inside media details. If we don't call onBackPressed then fragment stack is increasing every time.
+            //FIXME: Similar issue like this https://github.com/commons-app/apps-android-commons/issues/894
+            onBackPressed();
+        }
+        super.onResume();
     }
 
     /**

--- a/app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java
+++ b/app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java
@@ -137,6 +137,17 @@ public class SearchActivity extends NavigationBaseActivity implements MediaDetai
         mediaDetails.showImage(index);
     }
 
+    @Override
+    protected void onResume() {
+        if (supportFragmentManager.getBackStackEntryCount()==1){
+            //FIXME: Temporary fix for screen rotation inside media details. If we don't call onBackPressed then fragment stack is increasing every time.
+            //FIXME: Similar issue like this https://github.com/commons-app/apps-android-commons/issues/894
+            // This is called on screen rotation when user is inside media details. Ideally it should show Media Details but since we are not saving the state now. We are throwing the user to search screen otherwise the app was crashing.
+            // 
+            onBackPressed();
+        }
+        super.onResume();
+    }
 
     @Override
     public void onBackPressed() {

